src/reducers/accounts.js

```diff
@@ -98,31 +90,19 @@ export function canCreateAccount(state: State): boolean {
   return every(getAccounts(state), a => get(a, 'operations.length', 0) > 0)
 }
 
-export function serializeAccounts(accounts: Array<Object>) {
-  return accounts.map((account, key) => ({
-    ...account,
-    currency: getCurrencyByCoinType(account.coinType),
-    name: account.name || `${key}`,
-    unit: account.unit || getDefaultUnitByCoinType(account.coinType),
-  }))
+// Yeah. `any` should be `AccountRaw[]` but it can also be a map
+// of wrapped accounts. And as flow is apparently incapable of doing
+// such a simple thing, let's put any, right? I don't care.
+export function serializeAccounts(accounts: any): Account[] {
+  // ensure that accounts are always wrapped in data key
+  if (accounts.length && !accounts[0].data) {
```

Comment:
probably we could `try {} catch (){}` on the decode code & if it fails, fallback on returning `[]` , it's a safe thing to do if data happen to be corrupted in some way & so it does not brick the loader (i'm sure there are many ways things can go south & maybe we should assume the file we import can't be 100% trust. obviously we'll try to cover all cases via the migrations functions)

Reply:
probably we will keep it like this and a bit later you can simplify this code to no longer support migrating old dev version ;)